aptos-move/mvhashmap/src/versioned_aggregators.rs

```diff
@@ -15,13 +19,11 @@ use std::{
 // When an AggregatorEntry (see below) is transformed to an Estimate, internally we store
 // a potential bypass (based on the previously stored entry), which may allow a read
 // operation to not wait for the corresponding dependency.
-#[derive(Clone, Copy, Debug, PartialEq)]
+#[derive(Clone, Debug, PartialEq)]
 enum EstimatedEntry {
     NoBypass,
-    // If applicable, can bypass the Estimate by considering a Delta instead.
-    DeltaBypass(DeltaOp),
-    // If applicable, can bypass the Estimate by considering a Snapshot instead.
-    SnapshotBypass(AggregatorID, DeltaOp),
+    // If applicable, can bypass the Estimate by considering a apply change instead.
+    Bypass(AggregatorApplyChange),
```

Comment:
I don't have a strong opinion here. I am just wondering how much efficiency this is brining. 
Suppose `tx5` resulted in an error. Instead of using an Estimate to mark the aggregator in tx5, we are storing some change. Now, tx6 doesn't have to wait for tx5 and can use change. But later when tx5 re-runs, tx6 have re-run again as well. So, what's the benefit?

Reply:
if delta produced is the same, we don't need to re-execute.

So it's an optimization in case delta is always the same, i.e. independent from the rest of the computation.


as soon as a transaction produces a different delta, bypass is disabled for that aggregator - and we wait - as it is as you say - we will need to re-execute later.